### PR TITLE
minikube: set k8s version, use dedicated profile, add `make conbench-on-minikube` target for local dev

### PR DESCRIPTION
**Pull request by jgehrcke**
This is towards milestone 4 in https://github.com/conbench/conbench/issues/493#issuecomment-1413770175.

### CODE REVIEW

**Comment by jgehrcke** [comment]
CI failed with
```
Wait for port forwarding to take effect
The command curl --location --silent --output /dev/null http://127.0.0.1:8080/clusters timed out
==== FAILURE: OPERATOR DID NOT START OR PORT FORWARDING DID NOT WORK
If you observe issues with minikube VM not starting/not proceeding, consider deleting the .minikube dir and/or rebooting before re-running the script
Error: Process completed with exit code 1.
```
might be an instability, re-running.

Edit: yes, it's an instability -- tracking that here: https://github.com/conbench/conbench/issues/693.

File: Makefile [hunk 1]
```diff
@@ -131,20 +131,49 @@ build-conbench-container-image: set-build-info
...
-	minikube kubectl -- apply -f _build/deploy-conbench.yml
+	time minikube --profile "${MINIKUBE_PROFILE_NAME}" image load ${CONTAINER_IMAGE_SPEC}
+	minikube --profile "${MINIKUBE_PROFILE_NAME}" kubectl -- apply -f _build/deploy-conbench.yml
+
+
+# Thin wrapper currently not covered by CI. But the core
+# (../ci/minikube/test-conbench-on-mk.sh) is covered.
+.PHONY: conbench-on-minikube
+conbench-on-minikube: build-conbench-container-image start-minikube
+	rm -rf _build && mkdir -p _build && cd _build && bash ../ci/minikube/test-conbench-on-mk.sh
```
Comment (jgehrcke) [inline Makefile:157]: My plan is to make this the public-facing target. Documentation will come in another PR.

**Comment by jgehrcke** [comment]
Landing this, having the next PR building on top of this in the pipeline.

File: Makefile [hunk 2]
```diff
@@ -131,20 +131,49 @@ build-conbench-container-image: set-build-info
...
 
 
-# This target is used by ci/minikube/test-conbench-on-mk.sh. The `minikube
-# image load` technique allows for using local Docker images in k8s deployments
-# (as long as they specify `imagePullPolicy: Never`). That command however
-# takes a while for bigger images (about 1 min per GB, on my machine).
+# On GHA, the default minikube profile name is used, cannot be overridden yet
+# (see https://github.com/medyagh/setup-minikube/issues/59). Locally, we use a
+# specific profile name though. Note that this target here is invoked in the
+# context of ci/minikube/test-conbench-on-mk.sh. That wrapper is expected to set  The `minikube image load`
```
Comment (austin3dickey) [inline Makefile:137]: nit: remove `That wrapper is expected to set`

Comment (jgehrcke) [reply Makefile:137]: changed here: https://github.com/conbench/conbench/pull/725

**Pull request merged**